### PR TITLE
Perform bucket picking in web workers and other performance optimizations

### LINKED ISSUE #3823: Perform bucket picking in webworker
With the newest refactoring of the bucket pickers, it should be relatively easy to move the code into webworkers. Bucket picking is one of the top candidates for blocking the main thread afaik. So, the perf win  should be quite lucrative.

### PR DESCRIPTION
**Pull request by philippotto**
TLDR:
- Bucket picking is done in a dedicated webworker (bucket picking for selecting the correct magnification is still done synchronously, since the complexity wouldn't be worth it). The look-up from the bucket positions to the actual buckets is still done in the main thread, since we don't have easy access to the DataCube in the webworker.
- There is only one webworker which deals with all bucket picking tasks. I implemented a LatestTaskExecutor class which is used in combination with the bucket picking to ensure that bucket picking tasks cannot queue up (only the newest one is relevant). I tested this by slowing each bucket picking artificially done to take 200 ms.
- The results of the latest bucket picking result is shared across all data layers.
- I also ensured that bucket picking is not initiated for segmentation layers in arbitrary mode, even though this perf gain is bit overshadowed by the previous optimization.

A word abut the look up buffer for the TextureBucketManager:
The creation of that buffer is quite complex now and frankly a bit of a shit show. Since the bucket picking is done asynchronously, there are frames in which the TextureBucketManager doesn't know which buckets are supposed to be on the GPU. In that case, all available buckets should be put into the look up buffer, so that the shader can access them. I wrote a comment into the texture bucket which hopefully explains this. If it's unclear, hit me up. In the long-term I think we should do the addressing with cuckoo hashing or something similar.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- use ortho and the two flight modes
- test progressive vs. best quality loading strategy

### Issues:
- fixes #3823 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/model/bucket_data_handling/bucket_picker_strategies/flight_bucket_picker.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default function determineBucketsForFlight(
   const cameraDirection = V3.sub(centerPosition, cameraPosition);
   V3.scale(cameraDirection, 1 / Math.abs(V3.length(cameraDirection)), cameraDirection);
 
-  const iterStep = 10;
+  const iterStep = 8;
```
Comment (philippotto) [inline frontend/javascripts/oxalis/model/bucket_data_handling/bucket_picker_strategies/flight_bucket_picker.js]: I also tweaked the values for the arbitrary mode pickers a bit, since they are not that perf criticial anymore.

File: frontend/javascripts/test/libs/latest_task_executor.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  await scheduledPromise1.then(result => {
+    t.is(result, 1);
+  });
+  await scheduledPromise2.then(result => {
+    t.is(result, 2);
+  });
+});
+
+test("LatestTaskExecutor: three interleaving tasks", async t => {
```
Comment (daniel-wer) [inline frontend/javascripts/test/libs/latest_task_executor.spec.js]: Great tests! :+1:
